internal/internal_event_handlers.go

```diff
@@ -1379,3 +1384,139 @@ func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecut
 
 	return nil
 }
+
+func (weh *workflowExecutionEventHandlerImpl) estimateHistorySize(event *m.HistoryEvent) int {
```

Comment:
This function doesn't need to be attached to `*workflowExecutionEventHandlerImpl`. it only uses `weh.logger` which can be passed as argument. So please make this a standalone function and move to internal_utils.go as well as other helper functions you added below. This file is already huge (like many others in internal package unfortunately) and we will slowly refactor each type into its own file.

Reply:
Moved in latest commit